src/Component/Carousel.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * Class Carousel.
+ *
+ * @package Amp\AmpWP
+ */
+
+namespace Amp\AmpWP\Component;
+
+use DOMDocument;
+use DOMElement;
+use AMP_DOM_Utils;
+
+/**
+ * Class Carousel
+ *
+ * Gets the markup for an <amp-carousel>.
+ *
+ * @internal
+ * @since 1.5.0
+ */
+class Carousel {
+
+	/**
+	 * Value used for width of amp-carousel.
+	 *
+	 * @var int
+	 */
+	const FALLBACK_WIDTH = 600;
+
+	/**
+	 * Value used for height of amp-carousel.
+	 *
+	 * @var int
+	 */
+	const FALLBACK_HEIGHT = 480;
+
+	/**
+	 * An object representation of the DOM.
+	 *
+	 * @var DOMDocument
+	 */
+	protected $dom;
+
+	/**
+	 * The images to add to the carousel.
+	 *
+	 * @var DOMElementList
+	 */
+	protected $images;
+
+	/**
+	 * Instantiates the class.
+	 *
+	 * @param DOMDocument    $dom    The dom to use to create a carousel.
+	 * @param DOMElementList $images The images from which to create a carousel.
+	 */
+	public function __construct( $dom, DOMElementList $images ) {
+		$this->dom    = $dom;
+		$this->images = $images;
```

Comment:
As the carousel can contain other elements than just images, should we rename this property and the corresponding variables into `$slides` instead to match the `CaptionedSlide` adjustment?

Reply:
Ah, good point. It's now changed in https://github.com/ampproject/amp-wp/pull/3659/commits/450f9cdf106074db9a941ebcf0af8b91bcbd53d8

I should have [brought this up before](https://github.com/ampproject/amp-wp/pull/3659#discussion_r349240797): the `Carousel` class probably [makes the most sense for images](https://github.com/ampproject/amp-wp/pull/3659/files#diff-5d38610362507a06dc80a89b9151dff7R90-R98), and it's now only used for images.

But as you mentioned, the `DOMElementList` can now have any element.